dotcom-rendering/src/components/MostViewedRight.tsx

```diff
@@ -29,47 +30,115 @@ interface Props {
 	stickToTop?: boolean;
 }
 
+interface MostViewedRightPayloadType {
+	tabs: FETrailTabType[];
+}
+
 export const MostViewedRight = ({
 	limitItems = 5,
 	stickToTop = false,
 }: Props) => {
+	type Variant =
+		| 'deeply-read-only'
+		| 'deeply-read-and-most-viewed'
+		| 'most-viewed-only'
+		| 'none';
+	const useDeeplyReadTestVariant = (): Variant => {
+		const ABTestAPI = useAB()?.api;
+		if (
+			ABTestAPI?.isUserInVariant(
+				'DeeplyReadRightColumn',
+				'deeply-read-only',
+			)
+		) {
+			return 'deeply-read-only';
+		} else if (
+			ABTestAPI?.isUserInVariant(
+				'DeeplyReadRightColumn',
+				'deeply-read-and-most-viewed',
+			)
+		) {
+			return 'deeply-read-and-most-viewed';
+		} else if (
+			ABTestAPI?.isUserInVariant(
+				'DeeplyReadRightColumn',
+				'most-viewed-only',
+			)
+		) {
+			return 'most-viewed-only';
+		} else {
+			return 'none';
+		}
+	};
+	const testVariant = useDeeplyReadTestVariant();
```

Comment:
I think this looks like a custom hook but in practice it does not behave like one. The pattern should probably be a `useState` with a `useEffect` hook that depends on the AB test API.